src/main/java/com/amit/electronic/store/controller/UserController.java

```diff
@@ -0,0 +1,73 @@
+package com.amit.electronic.store.controller;
+
+import com.amit.electronic.store.entity.User;
+import com.amit.electronic.store.service.UserServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/store")
+public class UserController {
+
+    private final UserServiceImpl userServiceImpl;
+
+    UserController(@Autowired UserServiceImpl userServiceImpl) {
+        this.userServiceImpl = userServiceImpl;
+    }
+
+    @GetMapping
+    public String testHome() {
+        return "Welcome to Electronic Store, fellas!";
+    }
+
+    // create user endpoint
+    @PostMapping("/create")
+    public User createUser(@RequestBody User user) {
+        return userServiceImpl.create(user);
+    }
+
+    // get single user by id endpoint
+    @GetMapping("/get/user/id/{id}")
```

Comment:
All of the methods like getUserById, getUserByEmail, getUserByNameAndEmail, must be served from a single endpoint get_user. 
Introduce a GetUserRequest class with the necessary inputs and let the service class decide which JDBC implementation to call.
The JDBC constructs do not need to be exposed from the controller.

Reply:
This should be "/user"

Reply:
done

Reply:
![image](https://github.com/GitWizAmit/electronic-store-backend/assets/150045693/d3bcc1e0-c5ca-449b-946c-2a465fb198bf)